### PR TITLE
eLinks : Data model changes for eLinks V5.0

### PR DESCRIPTION
**Pull request by prudhvi-maddineni**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-6883

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/resources/db/migration/V1_65__elinks_alter_tables.sql [hunk 1]
```diff
@@ -0,0 +1,40 @@
+--Alter dbjudicialdata.judicial_user_profile
+ALTER TABLE dbjudicialdata.judicial_user_profile ADD COLUMN deleted_flag boolean;
+ALTER TABLE dbjudicialdata.judicial_user_profile ADD COLUMN date_of_deletion TIMESTAMP;
+
+--Alter dbjudicialdata.judicial_office_appointment
+ALTER TABLE dbjudicialdata.judicial_office_appointment ALTER COLUMN base_location_id SET NOT NULL;
+ALTER TABLE dbjudicialdata.judicial_office_appointment ALTER COLUMN appointment SET NOT NULL;
+ALTER TABLE dbjudicialdata.judicial_office_appointment ALTER COLUMN contract_type_id SET NOT NULL;
+
+--Alter dbjudicialdata.judicial_office_authorisation
```
Comment (manukundloo-hmcts) [inline src/main/resources/db/migration/V1_65__elinks_alter_tables.sql:10]: Please change the column name as mentioned in the ticket. It is juridiction only.

File: src/main/resources/db/migration/V1_65__elinks_alter_tables.sql [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+--Alter dbjudicialdata.judicial_office_appointment
+ALTER TABLE dbjudicialdata.judicial_office_appointment  DROP CONSTRAINT cft_region_id_fk1;
+ALTER TABLE dbjudicialdata.judicial_office_appointment RENAME COLUMN cft_region_id TO hmcts_region_id;
+ALTER TABLE dbjudicialdata.judicial_office_appointment ALTER COLUMN hmcts_region_id TYPE varchar(64);
+
+--create dbjudicialdata.hmcts_region_type
+CREATE TABLE dbjudicialdata.hmcts_region_type (
+	hmcts_region_id varchar(64) NOT NULL,
+	hmcts_region_desc_en varchar(256) NOT NULL,
+	hmcts_region_desc_cy varchar(256) NULL,
```
Comment (manukundloo-hmcts) [inline src/main/resources/db/migration/V1_65__elinks_alter_tables.sql:28]: Loke like NULL is the default value so you can remove it from line no 28

Comment (prudhvi-maddineni) [reply src/main/resources/db/migration/V1_65__elinks_alter_tables.sql]: Needful done @manukundloo-hmcts

File: src/main/resources/db/migration/V1_65__elinks_alter_tables.sql (hunk 1 above)
Comment (prudhvi-maddineni) [reply src/main/resources/db/migration/V1_65__elinks_alter_tables.sql:10]: Needful Done @manukundloo-hmcts

**Review by vasthav-patta — APPROVED** [review]

**Review by arshinsalim — APPROVED** [review]

**Pull request merged**